nucypher/blockchain/eth/actors.py

```diff
@@ -585,6 +594,83 @@ def encrypt_threshold_decryption_response(
             requester_public_key=requester_public_key,
         )
 
+
+    def _authorize_ritual(self, decryption_request: ThresholdDecryptionRequest):
+        # TODO: #3052 consider using the DKGStorage cache instead of the coordinator agent
+        # dkg_public_key = this_node.dkg_storage.get_public_key(decryption_request.ritual_id)
+        ritual = self.coordinator_agent.get_ritual(
+            decryption_request.ritual_id, with_participants=True
+        )
+
+        # enforces that the node is part of the ritual
+        participants = [p.provider for p in ritual.participants]
+        if self.checksum_address not in participants:
+            raise self.UnauthorizedRequest(
```

Comment:
Perhaps we separate this case from the case when the encryptor is not authorized by using a different exception?

Same for expired ritual?

Reply:
Unless we're going to raise a different HTTP status codes for those situations I don't think expanding the number of exception classes will have any benefit.  Expired rituals and threshold decryption requests sent to the wrong node are correctly understood as  *Unauthorized*, also the error message will be passed along with the HTTP response to hint to the requester what went wrong (ie 403 `Encrypted data not authorized for ritual x` )